app/controllers/admin/campaigns_controller.rb

```diff
@@ -37,4 +35,19 @@ def filter_campaign
                          .filter_by_status(params[:status])
                          .filter_by_creator(params[:creator])
   end
+
+  def handle_update_status success, notify_body
+    if success
+      flash[:success] = t ".update.success_change_status"
+
+      @campaign.notify_to_user(
```

Comment:
dùng callback_fuc để đảm bảo transaction

Reply:
Ở đây đang là campaign_controller của layout user, phía ngoài em cũng có 1 campaign_controller chung nữa. Cả 2 đều có công việc liên quan tới update campaign, em đã cho callback vào model Campaign ở cái update chung, còn với cái của riêng admin này em chưa nghĩ xem làm sao check được việc update từ chỗ nào mà ra. Ở đây phía dưới em có check biến success rồi sau đó mới bắn notify, em thấy nó vẫn đảm bảo tính toàn vẹn của transaction. Nên em xin phép giữ nguyên ạ (Đã được anh Hiệp đồng ý ạ)